airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3ConsumerFactory.java

```diff
@@ -0,0 +1,169 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3ConsumerFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(S3ConsumerFactory.class);
+  private static final DateTime SYNC_DATETIME = DateTime.now(DateTimeZone.UTC);
+
+  public AirbyteMessageConsumer create(final Consumer<AirbyteMessage> outputRecordCollector,
+                                       final BlobStorageOperations storageOperations,
+                                       final NamingConventionTransformer namingResolver,
+                                       final CheckedBiFunction<AirbyteStreamNameNamespacePair, ConfiguredAirbyteCatalog, SerializableBuffer, Exception> onCreateBuffer,
+                                       final JsonNode config,
+                                       final ConfiguredAirbyteCatalog catalog) {
+    final List<WriteConfig> writeConfigs = createWriteConfigs(storageOperations, namingResolver, config, catalog);
+    return new BufferedStreamConsumer(
+        outputRecordCollector,
+        onStartFunction(storageOperations, writeConfigs),
+        new SerializedBufferingStrategy(
+            onCreateBuffer,
+            catalog,
+            flushBufferFunction(storageOperations, writeConfigs, catalog)),
+        onCloseFunction(storageOperations, writeConfigs),
+        catalog,
+        storageOperations::isValidData);
+  }
+
+  private static List<WriteConfig> createWriteConfigs(final BlobStorageOperations storageOperations,
+                                                      final NamingConventionTransformer namingResolver,
+                                                      final JsonNode config,
+                                                      final ConfiguredAirbyteCatalog catalog) {
+    return catalog.getStreams()
+        .stream()
+        .map(toWriteConfig(storageOperations, namingResolver, config))
+        .collect(Collectors.toList());
+  }
+
+  private static Function<ConfiguredAirbyteStream, WriteConfig> toWriteConfig(
+                                                                              final BlobStorageOperations storageOperations,
+                                                                              final NamingConventionTransformer namingResolver,
+                                                                              final JsonNode config) {
+    return stream -> {
+      Preconditions.checkNotNull(stream.getDestinationSyncMode(), "Undefined destination sync mode");
+      final AirbyteStream abStream = stream.getStream();
+      final String namespace = abStream.getNamespace();
+      final String streamName = abStream.getName();
+      final String outputNamespace = getOutputNamespace(abStream, config.get("s3_bucket_path").asText(), namingResolver);
```

Comment:
nit: I am seeing a lot of places where we use strings for config parameters (`"s3_bucket_path", "s3_path_format" `in this file). IMHO, we should be replacing all these with constants, especially when we use them more than once. A search for `config.get("` currently returns 840 instances in Java, which is concerning.